langchain/src/output_parsers/structured.ts

```diff
@@ -104,15 +104,23 @@ ${JSON.stringify(zodToJsonSchema(this.schema))}
       const json = text.includes("```")
         ? text.trim().split(/```(?:json)?/)[1]
         : text.trim();
+
       return await this.schema.parseAsync(JSON.parse(json));
     } catch (e) {
-      try {
-        return await this.schema.parseAsync(JSON.parse(text.trim()));
-      } catch (e2) {
-        throw new OutputParserException(
-          `Failed to parse. Text: "${text}". Error: ${e2}`,
-          text
-        );
+      // eslint-disable-next-line no-instanceof/no-instanceof
+      if (e instanceof SyntaxError && e.message.includes("JSON")) {
```

Comment:
Can't use `instanceof` as it's not available in all JS environments.

Reply:
Do you think it would be better to change this condition to something like `if (e && (e as Error).name !== "ZodError")`?

Reply:
Unfortunately minifiers can also mess with that 😬

It looks like `SyntaxError` is a native object though?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SyntaxError

So probably ok to do the instanceof check